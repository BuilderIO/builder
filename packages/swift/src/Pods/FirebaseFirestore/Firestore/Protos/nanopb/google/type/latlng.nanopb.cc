/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* Automatically generated nanopb constant definitions */
/* Generated by nanopb-0.3.9.8 */

#include "latlng.nanopb.h"

#include "Firestore/core/src/nanopb/pretty_printing.h"

namespace firebase {
namespace firestore {

using nanopb::PrintEnumField;
using nanopb::PrintHeader;
using nanopb::PrintMessageField;
using nanopb::PrintPrimitiveField;
using nanopb::PrintTail;

/* @@protoc_insertion_point(includes) */
#if PB_PROTO_HEADER_VERSION != 30
#error Regenerate this file with the current version of nanopb generator.
#endif



const pb_field_t google_type_LatLng_fields[3] = {
    PB_FIELD(  1, DOUBLE  , SINGULAR, STATIC  , FIRST, google_type_LatLng, latitude, latitude, 0),
    PB_FIELD(  2, DOUBLE  , SINGULAR, STATIC  , OTHER, google_type_LatLng, longitude, latitude, 0),
    PB_LAST_FIELD
};


/* On some platforms (such as AVR), double is really float.
 * These are not directly supported by nanopb, but see example_avr_double.
 * To get rid of this error, remove any double fields from your .proto.
 */
PB_STATIC_ASSERT(sizeof(double) == 8, DOUBLE_MUST_BE_8_BYTES)

std::string google_type_LatLng::ToString(int indent) const {
    std::string header = PrintHeader(indent, "LatLng", this);
    std::string result;

    result += PrintPrimitiveField("latitude: ", latitude, indent + 1, false);
    result += PrintPrimitiveField("longitude: ", longitude, indent + 1, false);

    bool is_root = indent == 0;
    if (!result.empty() || is_root) {
      std::string tail = PrintTail(indent);
      return header + result + tail;
    } else {
      return "";
    }
}

}  // namespace firestore
}  // namespace firebase

/* @@protoc_insertion_point(eof) */
