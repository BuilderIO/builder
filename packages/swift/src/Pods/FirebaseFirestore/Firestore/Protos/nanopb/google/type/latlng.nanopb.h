/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* Automatically generated nanopb header */
/* Generated by nanopb-0.3.9.8 */

#ifndef PB_GOOGLE_TYPE_LATLNG_NANOPB_H_INCLUDED
#define PB_GOOGLE_TYPE_LATLNG_NANOPB_H_INCLUDED
#include <pb.h>

#include <string>

namespace firebase {
namespace firestore {

/* @@protoc_insertion_point(includes) */
#if PB_PROTO_HEADER_VERSION != 30
#error Regenerate this file with the current version of nanopb generator.
#endif


/* Struct definitions */
typedef struct _google_type_LatLng {
    double latitude;
    double longitude;

    std::string ToString(int indent = 0) const;
/* @@protoc_insertion_point(struct:google_type_LatLng) */
} google_type_LatLng;

/* Default values for struct fields */

/* Initializer values for message structs */
#define google_type_LatLng_init_default          {0, 0}
#define google_type_LatLng_init_zero             {0, 0}

/* Field tags (for use in manual encoding/decoding) */
#define google_type_LatLng_latitude_tag          1
#define google_type_LatLng_longitude_tag         2

/* Struct field encoding specification for nanopb */
extern const pb_field_t google_type_LatLng_fields[3];

/* Maximum encoded size of messages (where known) */
#define google_type_LatLng_size                  18

/* Message IDs (where set with "msgid" option) */
#ifdef PB_MSGID

#define LATLNG_MESSAGES \


#endif

}  // namespace firestore
}  // namespace firebase

/* @@protoc_insertion_point(eof) */

#endif
