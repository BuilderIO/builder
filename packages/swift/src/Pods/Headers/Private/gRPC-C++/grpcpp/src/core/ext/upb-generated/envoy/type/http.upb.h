../../../../../../../../../../gRPC-C++/src/core/ext/upb-generated/envoy/type/http.upb.h