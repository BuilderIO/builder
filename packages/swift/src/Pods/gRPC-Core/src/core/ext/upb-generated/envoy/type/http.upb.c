/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/http.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/http.upb.h"
#else
  #include  "envoy/type/http.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

