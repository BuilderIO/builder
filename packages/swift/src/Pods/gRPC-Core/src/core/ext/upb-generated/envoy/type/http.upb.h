/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/http.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#ifndef ENVOY_TYPE_HTTP_PROTO_UPB_H_
#define ENVOY_TYPE_HTTP_PROTO_UPB_H_

#if COCOAPODS==1
  #include  "third_party/upb/upb/generated_util.h"
#else
  #include  "upb/generated_util.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/decode.h"
#else
  #include  "upb/decode.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/encode.h"
#else
  #include  "upb/encode.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

#ifdef __cplusplus
extern "C" {
#endif

typedef enum {
  envoy_type_HTTP1 = 0,
  envoy_type_HTTP2 = 1,
  envoy_type_HTTP3 = 2
} envoy_type_CodecClientType;


#ifdef __cplusplus
}  /* extern "C" */
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

#endif  /* ENVOY_TYPE_HTTP_PROTO_UPB_H_ */
