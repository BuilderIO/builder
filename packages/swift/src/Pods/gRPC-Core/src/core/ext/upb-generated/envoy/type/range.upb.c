/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/range.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/range.upb.h"
#else
  #include  "envoy/type/range.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

static const upb_msglayout_field envoy_type_Int64Range__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 3, 1},
  {2, UPB_SIZE(8, 8), 0, 0, 3, 1},
};

const upb_msglayout envoy_type_Int64Range_msginit = {
  NULL,
  &envoy_type_Int64Range__fields[0],
  UPB_SIZE(16, 16), 2, false,
};

static const upb_msglayout_field envoy_type_Int32Range__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 5, 1},
  {2, UPB_SIZE(4, 4), 0, 0, 5, 1},
};

const upb_msglayout envoy_type_Int32Range_msginit = {
  NULL,
  &envoy_type_Int32Range__fields[0],
  UPB_SIZE(8, 8), 2, false,
};

static const upb_msglayout_field envoy_type_DoubleRange__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 1, 1},
  {2, UPB_SIZE(8, 8), 0, 0, 1, 1},
};

const upb_msglayout envoy_type_DoubleRange_msginit = {
  NULL,
  &envoy_type_DoubleRange__fields[0],
  UPB_SIZE(16, 16), 2, false,
};

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

