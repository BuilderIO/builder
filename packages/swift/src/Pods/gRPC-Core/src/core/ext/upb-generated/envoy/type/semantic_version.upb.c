/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/semantic_version.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/semantic_version.upb.h"
#else
  #include  "envoy/type/semantic_version.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

static const upb_msglayout_field envoy_type_SemanticVersion__fields[3] = {
  {1, UPB_SIZE(0, 0), 0, 0, 13, 1},
  {2, UPB_SIZE(4, 4), 0, 0, 13, 1},
  {3, UPB_SIZE(8, 8), 0, 0, 13, 1},
};

const upb_msglayout envoy_type_SemanticVersion_msginit = {
  NULL,
  &envoy_type_SemanticVersion__fields[0],
  UPB_SIZE(12, 12), 3, false,
};

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

