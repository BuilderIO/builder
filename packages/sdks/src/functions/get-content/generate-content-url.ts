import { flatten } from '../../helpers/flatten.js';
import {
  getBuilderSearchParamsFromWindow,
  normalizeSearchParams,
} from '../get-builder-search-params/index.js';
import type { GetContentOptions } from './types.js';

export const generateContentUrl = (options: GetContentOptions): URL => {
  const {
    limit = 30,
    userAttributes,
    query,
    noTraverse = false,
    model,
    apiKey,
    includeRefs = true,
    locale,
    apiVersion = 'v2',
  } = options;

  if (!apiKey) {
    throw new Error('Missing API key');
  }

  if (!['v2', 'v3'].includes(apiVersion)) {
    throw new Error(
      `Invalid apiVersion: expected 'v2' or 'v3', received '${apiVersion}'`
    );
  }

  const url = new URL(
    `https://cdn.builder.io/api/${apiVersion}/content/${model}?apiKey=${apiKey}&limit=${limit}&noTraverse=${noTraverse}&includeRefs=${includeRefs}${
      locale ? `&locale=${locale}` : ''
    }`
  );

  const queryOptions = {
    ...getBuilderSearchParamsFromWindow(),
    ...normalizeSearchParams(options.options || {}),
  };

  const flattened = flatten(queryOptions);
  for (const key in flattened) {
    url.searchParams.set(key, String(flattened[key]));
  }

  if (userAttributes) {
    url.searchParams.set('userAttributes', JSON.stringify(userAttributes));
  }
  if (query) {
    const flattened = flatten({ query });
    for (const key in flattened) {
      url.searchParams.set(key, JSON.stringify((flattened as any)[key]));
    }
  }

  return url;
};
