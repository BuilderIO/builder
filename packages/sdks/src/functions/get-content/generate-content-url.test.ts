import { generateContentUrl } from './generate-content-url';
import type { GetContentOptions } from './types';

const testKey = 'YJIGb4i01jvw0SRdL5Bt';
const testModel = 'page';
const testId = 'c1b81bab59704599b997574eb0736def';

const options = {
  cachebust: 'true',
  noCache: 'true',
  'overrides.037948e52eaf4743afed464f02c70da4':
    '037948e52eaf4743afed464f02c70da4',
  'overrides.page': '037948e52eaf4743afed464f02c70da4',
  'overrides.page:/': '037948e52eaf4743afed464f02c70da4',
  preview: 'page',
};

describe('Generate Content URL', () => {
  test('generates the proper value for a simple query', () => {
    const output = generateContentUrl({
      apiKey: testKey,
      model: testModel,
      query: { id: testId },
    });
    expect(output).toMatchSnapshot();
  });

  test('Handles overrides correctly', () => {
    const output = generateContentUrl({
      apiKey: testKey,
      model: testModel,
      query: { id: testId },
      options,
    });
    expect(output).toMatchSnapshot();
  });

  test('generate content url with apiVersion as v2', () => {
    const output = generateContentUrl({
      apiKey: testKey,
      model: testModel,
      query: { id: testId },
      options,
      apiVersion: 'v2',
    });
    expect(output).toMatchSnapshot();
  });

  test('generate content url with apiVersion as v3', () => {
    const output = generateContentUrl({
      apiKey: testKey,
      model: testModel,
      query: { id: testId },
      options,
      apiVersion: 'v3',
    });
    expect(output).toMatchSnapshot();
  });

  test('throw error when trying to generate content url with apiVersion as v1', () => {
    expect(() => {
      generateContentUrl({
        apiKey: testKey,
        model: testModel,
        query: { id: testId },
        options,
        apiVersion: 'v1' as GetContentOptions['apiVersion'],
      });
    }).toThrow(`Invalid apiVersion: expected 'v2' or 'v3', received 'v1'`);
  });

  test('throw error when trying to generate content url with an invalid apiVersion value', () => {
    expect(() => {
      generateContentUrl({
        apiKey: testKey,
        model: testModel,
        query: { id: testId },
        options,
        apiVersion: 'INVALID_API_VERSION' as GetContentOptions['apiVersion'],
      });
    }).toThrow(
      `Invalid apiVersion: expected 'v2' or 'v3', received 'INVALID_API_VERSION'`
    );
  });
});
