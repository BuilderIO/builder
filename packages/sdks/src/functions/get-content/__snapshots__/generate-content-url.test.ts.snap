// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generate Content URL > Handles overrides correctly 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&cachebust=true&noCache=true&overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&overrides.page=037948e52eaf4743afed464f02c70da4&overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview=page&query.id=%22c1b81bab59704599b997574eb0736def%22"`;

exports[`Generate Content URL > generate content url with apiVersion as default 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&cachebust=true&noCache=true&overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&overrides.page=037948e52eaf4743afed464f02c70da4&overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview=page&query.id=%22c1b81bab59704599b997574eb0736def%22"`;

exports[`Generate Content URL > generate content url with apiVersion as v2 1`] = `"https://cdn.builder.io/api/v2/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&cachebust=true&noCache=true&overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&overrides.page=037948e52eaf4743afed464f02c70da4&overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview=page&query.id=%22c1b81bab59704599b997574eb0736def%22"`;

exports[`Generate Content URL > generate content url with apiVersion as v3 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&cachebust=true&noCache=true&overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&overrides.page=037948e52eaf4743afed464f02c70da4&overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview=page&query.id=%22c1b81bab59704599b997574eb0736def%22"`;

exports[`Generate Content URL > generate content url with enrich option not present 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true"`;

exports[`Generate Content URL > generate content url with enrich option true 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&enrich=true"`;

exports[`Generate Content URL > generates the proper value for a simple query 1`] = `"https://cdn.builder.io/api/v3/content/page?apiKey=YJIGb4i01jvw0SRdL5Bt&limit=30&noTraverse=false&includeRefs=true&query.id=%22c1b81bab59704599b997574eb0736def%22"`;
